modules/gis/src/lib/binary-to-geojson.ts

```diff
@@ -123,7 +156,8 @@ function deduceReturnType(dataArray): 'FeatureCollection' | 'Geometry' {
 
 /** Parse input binary data and return an array of GeoJSON Features */
 function parseFeatureCollection(dataArray): Feature[] {
-  const features: Feature[] = [];
+  const featuresById: Map<number, Feature> = new Map<number, Feature>();
```

Comment:
If we know all the ids are covered, all we have to do is to create an array of the right length.
```suggestion
  const featuresById = new Array<Feature>(globalFeatureIds.length).fill(null);
```
This map business would seem to affect perf and memory.